imageio/core/functions.py

```diff
@@ -281,19 +324,26 @@ def mimread(uri, format=None, memtest=True, **kwargs):
     format : str
         The format to use to read the file. By default imageio selects
         the appropriate for you based on the filename and its contents.
-    memtest : bool
-        If True (default), this function will raise an error if the
-        resulting list of images consumes over 256 MB of memory. This
-        is to protect the system using so much memory that it needs to
-        resort to swapping, and thereby stall the computer. E.g.
+    memtest : {int, float, str}
+        If truthy, this function will raise an error if the resulting
+        list of images consumes greater than the amount of memory specified.
+        This is to protect the system from using so much memory that it needs
+        to resort to swapping, and thereby stall the computer. E.g.
         ``mimread('hunger_games.avi')``.
+        If the argument is a number, that will be used as the threshold number
+        of bytes.
+        If the argument is a string, it will be interpreted as a memory size with
+        units (e.g. '1kB', '250MiB', '80Yb'). This is case-sensitive.
+        If True, resort to the default (for compatibility reasons).
+        Default: '256MB'
```

Comment:
I agree with making this configurable like this. How strongly do you feel about the smart string parsing? I think I would prefer less complexity and accept only bool and int. In the documentation we can make a note that using `40*2**20` matches 40 MiB.

Reply:
I think it would be beneficial - people don't often think about their RAM in bytes now that 8+GB are commonplace, and the more readable and writeable it is, the better (too easy to drop or add an extra `*1000`). That said, I accept that too much flexibility adds API surface.

I'd prefer to keep it (I feel the function is well-defined and well-documented enough to avoid pain points, as it goes through a dict rather than truly parsing for the units), but of course defer to you.

Reply:
An alternative would be this:

```python
def mimread(uri, memtest=True, memlimit=None, **kwargs):
    if memtest:
        if memlimit:
            nbyte_limit = to_nbytes(memlimit)
        else:
            nbyte_limit = to_nbytes(MEMTEST_DEFAULT_MIM)
    else:
        nbyte_limit = False

    ...
```

This leaves the existing API untouched (where technically the original proposal is a breaking change), and reduces the number of things one parameter is doing.

Reply:
> people don't often think about their RAM in bytes now that 8+GB are commonplace, and the more readable and writeable it is, the better (too easy to drop or add an extra *1000).

I agree. Let's do it this way.

Reply:
Just to clarify, going with the additional `memlimit` parameter?

Reply:
No, with you original approach. I rather introduce a small backward-incompatibility for people that did not use `True`/`False` than convolute the API with an extra param.

Reply:
I see you already implemented the other, sorry for not being more clear earlier!